core/lib/multivm/src/tracers/call_tracer/vm_latest/mod.rs

```diff
@@ -187,7 +182,7 @@ impl CallTracer {
 
         if current_call.near_calls_after > 0 {
             current_call.near_calls_after -= 1;
-            self.stack.push(current_call);
+            self.push_call_and_update_stats(current_call.farcall, current_call.near_calls_after);
```

Comment:
I don't think you have to update number of calls here. For Each push, you are adding number of near calls to the depth_prefix as a result for near call with length 3 you will add 6 to the depth.